### ISSUE TITLE
ar No such file or directory when added some dependencies

### ISSUE THREAD

**Issue by arkrost** [ndmitchell/ghcid#309]
Hi! I've done the following steps:

1) Create empty dir, ie `mkdir sample`
2) Go to it, ie `cd sample`
3) Init cabal project, ie `cabal init`
4) Added `containers` dependency `containers ^>= 0.6.2` to build depends
5) Run ghcid

I got the following output and the repl is not starting. Can someone elaborate what's going wrong?
I have this problem not only with containers but with some others.

I tried to find a solution by myself. It turned out that `cabal repl --ghc-options=-fno-code` leads to the same output as well. But `cabal repl --repl-options="-fno-code"` works fine.
```
...
ar: dist/build/Data/Containers/ListUtils.o: No such file or directory
ar: dist/build/Data/IntMap.o: No such file or directory
ar: dist/build/Data/IntMap/Lazy.o: No such file or directory
ar: dist/build/Data/IntMap/Strict.o: No such file or directory
ar: dist/build/Data/IntMap/Strict/Internal.o: No such file or directory
ar: dist/build/Data/IntMap/Internal.o: No such file or directory
ar: dist/build/Data/IntMap/Internal/Debug.o: No such file or directory
ar: dist/build/Data/IntMap/Merge/Lazy.o: No such file or directory
ar: dist/build/Data/IntMap/Merge/Strict.o: No such file or directory
ar: dist/build/Data/IntSet/Internal.o: No such file or directory
ar: dist/build/Data/IntSet.o: No such file or directory
ar: dist/build/Data/Map.o: No such file or directory
ar: dist/build/Data/Map/Lazy.o: No such file or directory
ar: dist/build/Data/Map/Merge/Lazy.o: No such file or directory
ar: dist/build/Data/Map/Strict/Internal.o: No such file or directory
ar: dist/build/Data/Map/Strict.o: No such file or directory
ar: dist/build/Data/Map/Merge/Strict.o: No such file or directory
ar: dist/build/Data/Map/Internal.o: No such file or directory
ar: dist/build/Data/Map/Internal/Debug.o: No such file or directory
ar: dist/build/Data/Set/Internal.o: No such file or directory
ar: dist/build/Data/Set.o: No such file or directory
ar: dist/build/Data/Graph.o: No such file or directory
ar: dist/build/Data/Sequence.o: No such file or directory
ar: dist/build/Data/Sequence/Internal.o: No such file or directory
ar: dist/build/Data/Sequence/Internal/Sorting.o: No such file or directory
ar: dist/build/Data/Tree.o: No such file or directory
ar: dist/build/Utils/Containers/Internal/BitUtil.o: No such file or directory
ar: dist/build/Utils/Containers/Internal/BitQueue.o: No such file or directory
ar: dist/build/Utils/Containers/Internal/StrictPair.o: No such file or directory
ar: dist/build/Utils/Containers/Internal/State.o: No such file or directory
ar: dist/build/Utils/Containers/Internal/StrictMaybe.o: No such file or directory
ar: dist/build/Utils/Containers/Internal/PtrEquality.o: No such file or directory
ar: dist/build/Utils/Containers/Internal/Coercions.o: No such file or directory
ar: dist/build/Utils/Containers/Internal/TypeError.o: No such file or directory
ar: dist/build/Data/Map/Internal/DeprecatedShowTree.o: No such file or directory
ar: dist/build/Data/IntMap/Internal/DeprecatedDebug.o: No such file or directory
cabal: Failed to build containers-0.6.2.1 (which is required by exe:5E from
5E-0.1.0.0). See the build log above for details.
```

**Comment by ndmitchell** [comment]
Which version of Cabal are you using?

**Comment by arkrost** [comment]
I'm on Mac OS Catalina.
cabal-install version 3.0.0.0
compiled using version 3.0.0.0 of the Cabal library
The Glorious Glasgow Haskell Compilation System, version 8.6.5

**Comment by ndmitchell** [comment]
So we've been using these flags for a while. I wonder if Cabal changed to break it? Either way, I've switched to --repl-options. Can you try master and see if that works for you?

**Comment by arkrost** [comment]
Yep, it works fine. Thanks

**Comment by ndmitchell** [comment]
Great, released as 0.8.4 - thanks for testing.
